File: 0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;
        Arrays.sort(strs);
        int len = strs[0].length();//1
        String ans = "";
        int last = strs.length-1;

        for(int i=0; i < len; ) {//i is character index//0<1
                    if(strs[0].charAt(i) == (strs[last].charAt(i))) {
                        ans += strs[0].charAt(i);
                    } else {
                        break;
                    }
                    i++;
        }
        return ans;
    }
}

File: 0014-longest-common-prefix/solution.swift
class TrieNode {
    var child = [Character: TrieNode]()
    var isEnd = false
}

class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        var ans = ""
        var root = TrieNode()

        //insert all words
        for word in strs {
            var curr = root
            for ch in word {
                if curr.child[ch] == nil {
                    curr.child[ch] = TrieNode()
                }
                curr = curr.child[ch]!
            }
            curr.isEnd = true
        }

        var curr = root
        while curr.child.count == 1 && !curr.isEnd {
            let key = curr.child.keys.first!
            ans.append(key)
            curr = curr.child[key]!
        }
        return ans
    }
}
